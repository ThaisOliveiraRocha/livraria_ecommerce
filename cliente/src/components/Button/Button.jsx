import React from "react";
import PropTypes from "prop-types";
import { ButtonComponent } from "./Button.styles";
import Texto from "../Text";

const Button = ({ children, background, onClick, width, className }) => (
  <ButtonComponent
    background={background}
    onClick={onClick}
    width={width}
    className={className}
  >
    <Texto>{children}</Texto>
  </ButtonComponent>
);

Button.propTypes = {
  // bla: PropTypes.string,
};

Button.defaultProps = {
  // bla: 'test',
};

export default Button;
