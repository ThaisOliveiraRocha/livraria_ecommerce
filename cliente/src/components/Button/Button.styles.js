import styled from "styled-components";

export const Test = styled.div`
  display: flex;
`;

export const ButtonComponent = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${({ background }) =>
    background ? background : "#003d99"};
  color: white;
  font-size: 14px;
  width: ${({ width }) => (width ? width : "150px")};
  padding: 15px;
  margin: 5px;

  border: none;
  border-radius: 3px;
  outline: none;

  cursor: pointer;
  &:active {
    transform: scale(0.9, 0.9);
  }

  &.margin-right-10{
    margin-right: 10px;
  }

  &.close{
    background-color: red;
    color: #fff;
    border-radius: 50%;
    border: 1px solid white;
    position: absolute;
    top: -10px;
    right: -10px;
    width: 30px;
    height: 30px;
    padding: 0;
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;
