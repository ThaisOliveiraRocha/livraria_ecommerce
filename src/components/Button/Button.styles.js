import styled from "styled-components";

export const Test = styled.div`
  display: flex;
`;

export const ButtonComponent = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: ${({ background }) =>
    background ? background : "#003d99"};
  font-family: Impact, Haettenschweiler, "Arial Narrow Bold", sans-serif;
  color: white;
  font-size: 14px;
  font-weight: 600;
  width: ${({ width }) => (width ? width : "170px")};
  padding: 15px;

  border: none;
  border-radius: 3px;
  outline: none;

  cursor: pointer;
  &:active {
    transform: scale(0.9, 0.9);
  }
`;
