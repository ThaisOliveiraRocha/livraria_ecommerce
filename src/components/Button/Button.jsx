import React from 'react';
import PropTypes from 'prop-types';
import { ButtonComponent } from './Button.styles';

const Button = ({children, background, onClick, width, className}) => (
    <ButtonComponent background={background} onClick={onClick} width={width} className={className}>{children}</ButtonComponent>
);

Button.propTypes = {
  // bla: PropTypes.string,
};

Button.defaultProps = {
  // bla: 'test',
};

export default Button;
