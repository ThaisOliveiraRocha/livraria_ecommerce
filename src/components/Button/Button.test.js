import React from "react";
import { shallow } from "enzyme";
import Button from "./Button";

describe("<Button />", () => {
  it("should render without crashing", () => {
    shallow(<Button />);
  });
  it("should render basic infos", () => {
    const wrapper = shallow(<Button />);
    expect(wrapper).toMatchSnapshot();
  });
  
});
