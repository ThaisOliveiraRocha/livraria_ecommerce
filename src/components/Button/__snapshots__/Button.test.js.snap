// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render basic infos 1`] = `<styled.button />`;
